eslint.config.mjs

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import prettierConfig from 'eslint-config-prettier';
+import config from '@zendeskgarden/eslint-config';
+import noticePlugin from '@zendeskgarden/eslint-config/plugins/notice.js';
+import reactPlugin from '@zendeskgarden/eslint-config/plugins/react.js';
+import typeScriptPlugin from '@zendeskgarden/eslint-config/plugins/typescript.js';
+import typescriptTypeCheckedPlugin from '@zendeskgarden/eslint-config/plugins/typescript-type-checked.js';
+
+export default [
+  ...config,
+  noticePlugin,
+  reactPlugin,
+  prettierConfig,
+  {
+    ignores: [
+      '**/node_modules/**',
+      '**/public/**',
+      '**/.cache/**',
+      '**/.github/**',
+      '**/react-components/**',
+      '**/graphql.config.js'
+    ]
+  },
+  {
+    rules: {
+      'jsx-a11y/prefer-tag-over-role': 'off',
+      'no-unused-vars': 'off',
```

Comment:
https://stackoverflow.com/questions/57802057/eslint-configuring-no-unused-vars-for-typescript

Reply:
Hmm, that should already be turned off by including https://github.com/zendeskgarden/eslint-config/blob/main/plugins/typescript.js

Reply:
Still got those errors when I removed `'no-unused-vars': 'off'` and I couldn't find a way around it. 😢  

```
/Users/florent.mathieu/zendeskgarden/website/src/examples/design/icons/SvgSearch.tsx
  59:23  error  'event' is defined but never used  no-unused-vars

/Users/florent.mathieu/zendeskgarden/website/src/examples/forms/checkbox/Indeterminate.tsx
  33:21  error  'state' is defined but never used   no-unused-vars
  33:49  error  'action' is defined but never used  no-unused-vars

/Users/florent.mathieu/zendeskgarden/website/src/examples/forms/file-upload/FileUpload.tsx
  111:35  error  'index' is defined but never used  no-unused-vars

/Users/florent.mathieu/zendeskgarden/website/src/layouts/Root/components/SearchInput.tsx
  165:13  error  'Window' is defined but never used  no-unused-vars

✖ 5 problems (5 errors, 0 warnings)
```



> [The last matching config always wins when there is a conflict.](https://eslint.org/blog/2022/08/new-config-system-part-2/#goodbye-extends%2C-hello-flat-cascade)

I'm not sure why overriding doesn't work as expected. My suspicion is that `@zendeskgarden/eslint-config` uses `babelParser` while `@zendeskgarden/eslint-config/plugins/typescript.js` uses `tseslint.parser` and the 2 don't "talk" to each other. 🤷‍♂️ 🤔

Any thoughts?